packages/api/src/controllers/stream.ts

```diff
@@ -1069,12 +1081,12 @@ function publishRecordingStartedHook(
 /**
  * We don't actually send the webhook here, but schedule an event after a timeout.
  */
-function publishDelayedRecordingReadyHook(
+async function publishDelayedRecordingReadyHook(
```

Comment:
Do we still want to delay this event, or may as well send it immediately? That assuming that this will be called with an official "end" event from Mist, not the current `/setactive` which is fallible (if that's the case, let's keep the delay)

Reply:
Also something else: you might want to take a look on the caller of this function. It might need to change as well (e.g. not sending one event for each child stream, fetching sessions based on `recordingSessionId`/`sessionId` field and not just assuming there is 1 session with the same ID of the child stream (1-1).

Reply:
Both these points are discussed in different comments.

1. There will be no "official end` event, so I think we should leave it as it is.
2. I think we'll still have 1-1 relation between child streams and sessions.

Reply:
I'm gonna check with @emranemran, I haven't seen this decision of keeping child streams and sessions 1-1 and I don't know how that's gonna work on the API/dashboard.